.github/workflows/scala.yml

```diff
@@ -1,17 +1,88 @@
-name: Scala CI
+name: Scala CI & Release
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ '*' ] # Trigger on all branches
+    tags:
+      - 'v*' # Trigger on version tags like v1.0.0
+  pull_request:
+    branches: [ master ] # Keep PR trigger for master
+  workflow_dispatch:
 
 jobs:
-  build:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # Required for sbt-dynver to access Git history
+    - name: Set up JDK 11
+      uses: actions/setup-java@v4
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+    - name: Setup sbt
+      uses: coursier/setup-action@v1
+      with:
+        apps: sbt
+    - name: Run tests
+      run: sbt +test
 
+  publish-snapshot:
+    name: Publish Snapshot
+    needs: test
+    if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write # Permission to write to GitHub Packages
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # Required for sbt-dynver
+    - name: Set up JDK 11
+      uses: actions/setup-java@v4
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+    - name: Setup sbt
+      uses: coursier/setup-action@v1
+      with:
+        apps: sbt
+    - name: Publish Snapshot to GitHub Packages
+      run: sbt +publish
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Use the default GitHub token for Packages auth
 
+  publish-release:
+    name: Publish Release
+    needs: test # Run after tests pass
+    if: github.ref_type == 'tag' && startsWith(github.ref, 'refs/tags/v') # Only run on version tags
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # Permission to create release
+      packages: write # Permission to write to GitHub Packages
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up JDK 1.8
-        uses: actions/setup-java@v1
-        with:
-          java-version: 1.8
-      - name: Run tests
-        run: sbt +test
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # Required for sbt-dynver
+    - name: Set up JDK 11
+      uses: actions/setup-java@v4
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+    - name: Setup sbt
+      uses: coursier/setup-action@v1
+      with:
+        apps: sbt
+    - name: Extract version from tag
```

Comment:
It should use maven release here. And only maven release (maven release should place the tag)